### ISSUE TITLE
Enhancements / ideas for an iNAV ESP32 companion module

### ISSUE THREAD

**Issue by nikil511** [DzikuVx/QmuBeacon#2]
- connect beacon to FC to use it's GPS/telemetry instead of (additional) GPS. Thus take advantage of ESP32+LoRa modules (e.g. HELTEC / TTGO LORA ESP32 with OLED) without GPS

take advantage of the FC connection to:
- display useful data to OLED (bat, channel, mission, etc)
- mission planning over ESP32 WiFi / BT

only problem with above ideas that a low cost FC will probably not have any spare UART.... but now that you have an ESP32 on the drone, you could create "Serial HUB"  
i.e. have a few serial (wrapper) UART ports on ESP32 that combine GPS / SmartAudio / Telemetry etc and connect to only one UART of the FC. Could be ESP32 native or softserial

So you are essentially multiplexing multiple UARTs in to one, and this would require a modification to iNAV to demultiplex, but opens up many possibilities. (?)

**Comment by DzikuVx** [comment]
Yes, this is also planned to get telemetry from flight controller
